dependency/actions/compile/action.yml

```diff
@@ -0,0 +1,54 @@
+name: 'Compile HTTPD on Target'
+description: |
+  Compiles HTTPD given a version, output directory, and a target image to compile on
+
+inputs:
+  version:
+    description: 'dependency version'
+    required: true
+  outputDir:
+    description: 'output directory'
+    required: true
+  target:
+    description: 'dependency OS target variant'
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+
+  - name: docker build
+    id: docker-build
+    env:
+      SKIP_LOGIN: true
+    uses: actions-hub/docker/cli@master
+    with:
+      args: "build -t compilation -f dependency/actions/compile/${{ inputs.target }}.Dockerfile dependency/actions/compile"
+
+  - name: docker run
+    id: docker-run
+    uses: actions-hub/docker/cli@master
+    env:
+      SKIP_LOGIN: true
+    with:
+      args: "run -v ${{ inputs.outputDir }}:/home compilation --version ${{ inputs.version }} --outputDir /home --target ${{ inputs.target }}"
+
+  - name: print contents of output dir
+    shell: bash
+    run: ls -lah ${{ inputs.outputDir }}
+
+  - name: prepare test
+    id: prepare-test
+    uses: actions-hub/docker/cli@master
+    env:
+      SKIP_LOGIN: true
+    with:
+      args: "build -t test -f dependency/test/${{ inputs.target }}/Dockerfile dependency/test/${{ inputs.target }}"
```

Comment:
What's the thinking with bringing in this `actions-hub/docker/cli` action? Can we not just run the `docker` CLI directly since its available on all GitHub runners?

Reply:
This was a copy/paste. I'll use the cli on the runner.